src/background/providers/cloudflare.ts

```diff
@@ -1,72 +1,90 @@
-import * as voprf from '../voprf';
+import * as voprf from '../crypto/voprf';
 
-import { Callbacks, Provider } from '.';
+import { Provider, EarnedTokenCookie, Callbacks, QUALIFIED_HOSTNAMES, QUALIFIED_PATHNAMES, QUALIFIED_PARAMS, isIssuingHostname, isQualifiedPathname, areQualifiedQueryParams, areQualifiedBodyFormParams } from './provider';
 import { Storage } from '../storage';
 import Token from '../token';
 import axios from 'axios';
 import qs from 'qs';
 
-const ISSUE_HEADER_NAME = 'cf-chl-bypass';
-const NUMBER_OF_REQUESTED_TOKENS = 30;
-const ISSUANCE_BODY_PARAM_NAME = 'blinded-tokens';
+const NUMBER_OF_REQUESTED_TOKENS: number = 30;
+const DEFAULT_ISSUING_HOSTNAME:   string = 'captcha.website';
+const CHL_BYPASS_SUPPORT:         string = 'cf-chl-bypass';
+const ISSUE_HEADER_NAME:          string = 'cf-chl-bypass';
+const ISSUANCE_BODY_PARAM_NAME:   string = 'blinded-tokens';
 
-const COMMITMENT_URL =
+const COMMITMENT_URL: string =
     'https://raw.githubusercontent.com/privacypass/ec-commitments/master/commitments-p256.json';
 
-const QUALIFIED_QUERY_PARAMS = ['__cf_chl_captcha_tk__', '__cf_chl_managed_tk__'];
-const QUALIFIED_BODY_PARAMS = ['g-recaptcha-response', 'h-captcha-response', 'cf_captcha_kind'];
-
-const CHL_BYPASS_SUPPORT = 'cf-chl-bypass';
-const DEFAULT_ISSUING_HOSTNAME = 'captcha.website';
+const ALL_ISSUING_CRITERIA: {
```

Comment:
Making the criteria programmable is more flexible than making it configurable. Please read https://blog.cloudflare.com/privacy-pass-v3/ for more detail. We had difficultly by making things configurable in the past because configuration is less flexible.

That is, if we want to change the criteria, it makes more sense to change the code in some function rather change some configuration object.

Reply:
yes and no..

it would be bad to force the use of a convention by all providers; however, since some tasks are repeated by all providers.. it makes sense to me to provide tooling in the form of static methods that providers have the option to use.

Those tools will of course require formatted input that providers who choose to call these methods will need to supply; that all that these data structures are for.

Reply:
to make my earlier comment more concrete..

`handleBeforeRequest()` needs to _quickly_ determine whether the intercepted request:
1. solves a capcha belonging to the particular provider
2. is being solved on a domain that issues tokens

these determinations are always based on:
1. the requested url
2. POST data (if present)

I added static methods that allow providers to pass the data from the request as well as a configuration object.. and the result is a simple yes or no.. the request does or doesn't match the filter criteria as configured

If a provider wants to apply specialized logic.. then that's great; simply don't use these methods and roll your own.